kroxylicious-app/src/assembly/kroxylicious-start.sh

```diff
@@ -25,10 +27,51 @@ classpath() {
   echo "${class_path}"
 }
 
+# Disable warnings about `local` variables
+# shellcheck disable=SC3043
+native_library_path() {
+  NATIVE_LIB_BASE_DIR="/opt/kroxylicious/libs/native/"
+  local lib_path="${1}"
+  arch="${TARGETARCH:-$(uname -m)}"
+  native_lib="${NATIVE_LIB_BASE_DIR}${lib_path}/${TARGETOS:-linux}"
+  if [ -r "${native_lib}" ]; then
+    if [ -r "${native_lib}/${arch}" ]; then
+      echo "${native_lib}/${arch}"
+      return
+    else
+      # not everyone thinks `uname -m` is the right convention
+      case ${arch} in
+        arm64) arch=aarch64 ;;
+        x86_64) arch=amd64 ;;
+      esac
+      if [ -r "${native_lib}/${arch}" ]; then
+          echo "${native_lib}/${arch}"
+          return
+      fi
+    fi
+  elif [ -r "${NATIVE_LIB_BASE_DIR}${lib_path}/" ]; then
+    # The native dependency isn't broken down by OS and arch but still has something loadable. (Looking at you Netty)
```

Comment:
Hmm comment is a bit miss leading as Netty does break things down by OS and arch but it doesn't do it in the directory structure, it does it in the filename.